autoconnect/autoconnect-ws/src/handler.rs

```diff
@@ -0,0 +1,291 @@
+use std::sync::Arc;
+
+use actix_ws::{CloseReason, Message};
+use futures::{channel::mpsc, stream::StreamExt};
+use tokio::{
+    select,
+    time::{interval, timeout},
+};
+
+use autoconnect_common::protocol::{ServerMessage, ServerNotification};
+use autoconnect_settings::AppState;
+use autoconnect_ws_sm::{UnidentifiedClient, WebPushClient};
+
+use crate::{
+    error::WSError,
+    session::{Session, SessionImpl},
+};
+
+/// WebPush WebSocket handler Task
+pub async fn webpush_ws(
+    session: actix_ws::Session,
+    msg_stream: actix_ws::MessageStream,
+    app_state: Arc<AppState>,
+    ua: String,
+) {
+    let client = UnidentifiedClient::new(ua, app_state);
+    let mut session = SessionImpl::new(session.clone());
+    let close_reason = _webpush_ws(client, &mut session, msg_stream)
+        .await
+        .unwrap_or_else(|e| {
+            Some(CloseReason {
+                code: e.close_code(),
+                // TODO: clip this String to =~ 100, WS control frames (such as
+                // close) are limited to 125 bytes
+                description: Some(e.to_string()),
```

Comment:
Hrm, should we just code the description element rather than return an error string? This really isn't for human consumption, so we might be able to use a errno or phrase that can be correlated with the server log.